python/ray/data/context.py

```diff
@@ -11,8 +11,14 @@
 _context_lock = threading.Lock()
 
 # The max target block size in bytes for reads and transformations.
+# We choose 512MiB as 8x less than the typical memory:core ratio of 4:1.
 DEFAULT_TARGET_MAX_BLOCK_SIZE = 512 * 1024 * 1024
 
+# Datasets will avoid creating blocks smaller than this size in bytes on read.
+# This takes precedence over MIN_SAFE_PARALLELISM.
+# We choose 100KiB to be consistent with the Ray object inline threshold (100KiB).
+DEFAULT_TARGET_MIN_BLOCK_SIZE = 100 * 1024
```

Comment:
100 KiB seems too small for Ray's task and general object overhead, are we sure we don't want a minimum in the range of 1-128 MiB?

Reply:
I'd be ok with 1MiB, but note that this optimization is only reducing the overhead of tiny datasets (e.g., example datasets), shouldn't matter too much in practice.

Reply:
Ah, since any dataset larger than 20 MiB will be hitting the 200 default parallelism ceiling. I think that 1 MiB would be better, that should mitigate an odd-seeming number of read tasks being spanned for e.g. reading a 20 MiB dataset with 200 files on an 8 CPU core machine launching 200 read tasks.

Reply:
Alright, changed to 1MiB.